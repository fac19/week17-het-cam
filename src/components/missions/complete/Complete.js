import React from "react";
import { Link } from "react-router-dom";
import badge from "./badge.svg";
import circle from "./circle.svg";
import { ImageWrapper, Image, Circle, Text } from "./Complete.style";
import Confetti from "../../global/confetti/Confetti";
import { accentColour, linkColour } from "../../global/colours";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  links: {
    fontFamily: "Josefin Sans, sans-serif",
    transition: "all 0.2s ease-in-out",
    color: linkColour,
    textDecoration: "underline",
    fontSize: "1.2rem",
    "&:hover": {
      fontSize: "1.8rem",
      color: accentColour,
    },
    display: "block",
    textAlign: "center",
  },
});

const Complete = (props) => {
  const classes = useStyles();

  return (
    <>
      <Confetti />
      <Text>MISSION COMPLETE!!</Text>
      <ImageWrapper>
        <Circle src={circle} />
        <Image src={badge} />
      </ImageWrapper>
      <Link to="/my-missions" className={classes.links}>
        NEXT MISSION UNLOCKED
      </Link>
    </>
  );
};

export default Complete;
