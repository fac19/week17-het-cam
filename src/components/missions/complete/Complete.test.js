import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Complete from "./Complete";

test("Mission complete page rendering correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Complete />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
