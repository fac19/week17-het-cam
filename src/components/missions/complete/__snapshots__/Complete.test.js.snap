// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission complete page rendering correctly 1`] = `
Array [
  <canvas
    height={768}
    initialY={16}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 2,
      }
    }
    width={1024}
  />,
  <h2
    className="sc-AxhCb kqvdZj"
  >
    MISSION COMPLETE!!
  </h2>,
  <section
    className="sc-AxjAm iSFHIL"
  >
    <img
      className="sc-AxiKw chpAef"
      src="circle.svg"
    />
    <img
      className="sc-AxirZ dwTxAr"
      src="badge.svg"
    />
  </section>,
  <a
    className="makeStyles-links-1"
    href="/my-missions"
    onClick={[Function]}
  >
    NEXT MISSION UNLOCKED
  </a>,
]
`;
